### ISSUE TITLE
MVC Sample - FrameworkReference & .NET Core 3 questions

### ISSUE THREAD

**Issue by zHaytam** [natemcmaster/DotNetCorePlugins#55]
Hi,

I've been checking the mvc sample you've provided and I was interested as to why and how you knew about the `Microsoft.NET.Sdk.Razor`, the `FrameworkReference` ([here](https://github.com/natemcmaster/DotNetCorePlugins/blob/master/samples/aspnetcore-mvc/MvcAppPlugin1/MvcAppPlugin1.csproj#L9))? 

After some research, I now understand what they mean. Using `Microsoft.AspNetCore.App` means you're including the full ASP.NET Core 3. What I don't get is that the Web application (.NET Core 3) doesn't have a `FrameworkReference` nor Nuget packages so how does dotnet know?

Thank you!

**Comment by natemcmaster** [comment]
@zHaytam I know about this stuff because I worked at Microsoft and worked with the group of devs that built the FrameworkReference feature.

In .NET Core 3.0, the "Sdk" listed on the `<Project>` element implies some `FrameworkReference` items, as follows:

Sdk | `<TargetFramework>` | Implicit `<FrameworkReference>`'s
--|--|--
Microsoft.NET.Sdk | netstandard2.1 | NETStandard.Library
Microsoft.NET.Sdk | netcoreapp3.0 | Microsoft.NETCore.App
Microsoft.NET.Sdk.Razor | netcoreapp3.0 | Microsoft.NETCore.App
Microsoft.NET.Sdk.Web | netcoreapp3.0 | Microsoft.NETCore.App<br /> Microsoft.AspNetCore.App


As you see, Microsoft.NET.Sdk.Razor doesn't implicitly reference aspnetcore, so it must be listed in the project file. I actually suggested a while ago that the `FrameworkReference` to aspnetcore should be implicit (see https://github.com/aspnet/AspNetCore/issues/4176), but the Razor team decided against it.
